File: 153-FindMinimumInRotatedSortedArray/153-FindMinimumInRotatedSortedArray.py
# Last updated: 6/11/2026, 11:35:38 AM
class Solution:
    def findMin(self, nums: List[int]) -> int:
        low=0
        high = len(nums)-1
        mid = (low+high)//2

        ans = 10000
        while(low<=high):
            mid = (low+high)//2
            ans = min(ans, nums[mid], nums[low], nums[high])

            if(nums[0]<nums[mid]):
                # ans = min(ans,nums[low]s,);           
                low = mid+1
                
            else:
                # ans =  min(ans, nums[high])
                high= mid-1
                

        return ans
        

File: 153-FindMinimumInRotatedSortedArray/153-FindMinimumInRotatedSortedArray.cpp
// Last updated: 6/11/2026, 11:35:32 AM
class Solution {
public:
    int findMin(vector<int>& arr) {
        int n = arr.size();
        int lo = 0;
        int hi = n-1;
        int ans = arr[0];
        while(lo<=hi){
            int mid = (lo+hi)/2;
            if(arr[mid]>=ans){
                lo = mid+1;
            }else{
                ans = arr[mid];
                hi = mid-1;
            }
        }

        return ans;
    }
};